app/src/main/java/com/concordia/mcga/fragments/BottomSheetBuildingInfoFragment.java

```diff
@@ -0,0 +1,399 @@
+            @Override
+            public void onStateChanged(@NonNull View bottomSheet, int newState) {
+                switch (newState) {
+                    case BuildingBottomSheetInfo.STATE_COLLAPSED:
+                        expandButton.setImageResource(R.drawable.ic_expand_less_black_24dp);
+                        break;
+
+                    case BuildingBottomSheetInfo.STATE_EXPANDED:
+                        expandButton.setImageResource(R.drawable.ic_expand_more_black_24dp);
+                        break;
+
+                    default:
+                        break;
+                }
+            }
+            @Override
+            public void onSlide(@NonNull View bottomSheet, float slideOffset) {
+            }
+        });
+    }
+
+    /**
+     * Instanciate the arraylist adapter and assign it to the list
+     */
+    private void setupListAdapter(){
+        adapter = new BuildingInformationArrayAdapter(getActivity().getApplicationContext(), images);
+        adapter.notifyDataSetChanged();
+        list.setAdapter(adapter);
+    }
+
+    /**
+     *  Override on click method to define it's behavior
+     * @param clickedView
+     */
+    // Overloaded method for button clicks
+    @Override
+    public void onClick(View clickedView) {
+        switch (clickedView.getId()) {
+            case R.id.expandButton:
+                if (behavior.getState() == BuildingBottomSheetInfo.STATE_COLLAPSED){
+                    behavior.setState(BuildingBottomSheetInfo.STATE_EXPANDED);
+                }else{
+                    behavior.setState(BuildingBottomSheetInfo.STATE_COLLAPSED);
+                }
+                break;
+        }
+    }
+
+    /**
+     * Add textual information into textboxes found in the layout.xml
+     * @param bottom_sheet_title
+     * @param address
+     * @param openingTime
+     * @param closingTime
+     */
+    // Set building info 1 shot easy money
+    public void setBuildingInformation(String bottom_sheet_title, String address, String openingTime, String closingTime){
+        setBottomSheetTitle(bottom_sheet_title);
+        setAddress(address);
+        setOpeningTime(openingTime);
+        setClosingTime(closingTime);
+    }
+
+    /**
+     * Forces the Bottom Sheet to collapse
+     */
+    public void collapse(){
+        behavior.setState(BuildingBottomSheetInfo.STATE_COLLAPSED);
+    }
+
+    /**
+     * Forces the Bottom Sheet to expand
+     */
+    public void expand(){
+        behavior.setState(BuildingBottomSheetInfo.STATE_EXPANDED);
+    }
+
+    /**
+     * Obtains the state of the Bottom Sheet
+     * 4 is expanded
+     * 5 is collapsed
+     * Others states should not exist
+     * @return
+     */
+    public int getState(){
+        return behavior.getState();
+    }
+
+    /**
+     * Updates the row of images associate to the building information. Works
+     * similarly to a unit of Work
+     */
+    public void updateImageRow(){
```

Comment:
Wouldn't this mean that some of the images you have are never shown? "--rowImages.get(2)---img---img[...]". Unsure if I understand your method properly.